src/Nager.Date/HolidaySystem.cs

```diff
@@ -135,6 +135,8 @@ public static class HolidaySystem
                 { CountryCode.ZW, new Lazy<IHolidayProvider>(() => new ZimbabweHolidayProvider(_catholicProvider))}
             };
 
+        private static bool _licenseValid = false;
```

Comment:
Should be of type `bool?`, so that even in the negative case it is only tested once.
```suggestion
        private static bool? _licenseValid;
```